### PR TITLE
Add trayicon on macOS and enable minimize onClose

### PR DESCRIPTION
**Pull request by TheZ3ro**
## Description

Closes https://github.com/keepassxreboot/keepassxc/issues/498 and https://github.com/keepassxreboot/keepassxc/issues/438

## How has this been tested?

Locally and with Travis

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

### CODE REVIEW

**Comment by starquake** [comment]
Is it possible to enable "minimize on close" but without a trayicon?

**Comment by phoerious** [comment]
Yes, that is already possible.

**Comment by starquake** [comment]
Sorry, let me rephrase the question. I can not seem to find a combination of settings to make it work like that. :-) 

How can I set it to minimize on close and not show a tray icon? 

:-)

**Comment by phoerious** [comment]
Actually, you're right. It's not entirely possible to have that behavior without at least showing the tray icon.

**Comment by starquake** [comment]
Okay, in that case I will report an issue.
